internal/scheduler/scheduling/queue_scheduler.go

```diff
@@ -112,15 +112,19 @@ func (sch *QueueScheduler) Schedule(ctx *armadacontext.Context) (*SchedulerResul
 		}
 		select {
 		case <-ctx.Done():
-			// TODO: Better to push ctx into next and have that control it.
-			err := ctx.Err()
-			sctx.TerminationReason = err.Error()
-			return nil, err
+			sctx.TerminationReason = ctx.Err().Error()
 		default:
 		}
+		if ctx.Err() != nil {
```

Comment:
Are you sure this is safe?  The scheduler has multiple stages (evict, reschedule, remove overfill) and I'm not sure that a timeout at an arbitrary point will result in a usable set of schedulable jobs.  The test below handles the simple case where
we are scheduling jobs onto an empty cluster with no preemption but I suspect other cases will fail:
- Fair share prremption
- Urgency based preemption (particularly in the case of overfill)